extensions/emmet/src/util.ts

```diff
@@ -128,6 +128,50 @@ export function parseDocument(document: vscode.TextDocument, showError: boolean
 	return undefined;
 }
 
+export function parsePartialStylesheet(document: vscode.TextDocument, position: vscode.Position): Node | undefined {
+
+	let startPosition = new vscode.Position(0, 0);
+	let endPosition = new vscode.Position(document.lineCount - 1, document.lineAt(document.lineCount - 1).text.length);
+	const closeBrace = 125;
+	const openBrace = 123;
+
+	// Go forward until we found a closing or opening brace
+	let stream = new DocumentStreamReader(document, position);
+	stream.eatWhile(char => { return char !== closeBrace && char !== openBrace; });
+	stream.eat(closeBrace);
+	endPosition = stream.pos;
+
+	// Go back until we found an opening brace. If we find a closing one, we first find its opening brace and then we continue.
+	stream.pos = position;
+	let openBracesRemaining = 1;
+	while (openBracesRemaining > 0 && !stream.sof()) {
+		let ch = stream.backUp(1);
+		if (ch === openBrace) {
+			openBracesRemaining--;
+		} else if (ch === closeBrace) {
+			if (document.languageId !== 'css') {
+				openBracesRemaining++;
+			}
+			if (position.line - stream.pos.line > 1000) {
+				return parseStylesheet(new DocumentStreamReader(document, new vscode.Position(0, 0)));
```

Comment:
This is missing the range to be passed

Reply:
Also, move this code to the beginning of the while loop. No reason to have it inside the if/else correct?